src/megaapi_impl.cpp

```diff
@@ -24975,9 +24985,12 @@ void MegaFolderUploadController::start(MegaNode*)
         return;
     }
 
-    auto localpath = LocalPath::fromPath(transfer->getPath(), *client->fsaccess);
-    scanFolder(parent->getHandle(), parent->getHandle(), localpath, folderName);
-    createFolder();
+    mThread = std::thread ([this]() {
+        auto localpath = LocalPath::fromPath(transfer->getPath(), *client->fsaccess);
+        scanFolder(transfer->getParentHandle(), transfer->getParentHandle(), localpath, transfer->getFileName());
+        createFolder();
```

Comment:
Mutual exclusion violated here. The `createFolder()` adds API commands to the internal queue of the SDK, which is not thread-safe.